Parsers/ASimAuditEvent/Parsers/vimAuditEventGithubWebhook.yaml

```diff
@@ -0,0 +1,121 @@
+      Link: https://aka.ms/AboutASIM
+    - Title: Microsoft GithubWebhook Events product documentation
+      Link: https://docs.github.com/en/webhooks/webhook-events-and-payloads
+Description: |
+    This ASIM parser supports filtering and normalizing Microsoft Github audit events in the Githubscanaudit_C.
+ParserName: vimAuditEventMicrosoftGithubWebhook
+EquivalentBuiltInParser: _Im_AuditEvent_MicrosoftGithubWebhook
+ParserParams:
+    - Name: starttime
+      Type: datetime
+      Default: datetime(null)
+    - Name: endtime
+      Type: datetime
+      Default: datetime(null)
+    - Name: srcipaddr_has_any_prefix
+      Type: dynamic
+      Default: dynamic([])
+    - Name: actorusername_has_any
+      Type: dynamic
+      Default: dynamic([])
+    - Name: operation_has_any
+      Type: dynamic
+      Default: dynamic([])
+    - Name: eventtype_in
+      Type: dynamic
+      Default: dynamic([])
+    - Name: eventresult
+      Type: string
+      Default: "*"
+    - Name: object_has_any
+      Type: dynamic
+      Default: dynamic([])
+    - Name: newvalue_has_any
+      Type: dynamic
+      Default: dynamic([])
+    - Name: disabled
+      Type: bool
+      Default: false
+ParserQuery: |
+let parser = (
+    starttime:datetime=datetime(null), 
+    endtime:datetime=datetime(null), 
+    username_has_any:dynamic = dynamic([]),
+    targetappname_has_any:dynamic = dynamic([]),
+    srcipaddr_has_any_prefix:dynamic = dynamic([]),
+    srchostname_has_any:dynamic = dynamic([]),
+    eventtype_in:dynamic = dynamic([]),
+    eventresultdetails_in:dynamic = dynamic([]),
+    eventresult:string = '*',
+    disabled:bool=false
+)
+{
+     let EventResultDetailslookup=datatable(EventOriginalType:string,EventResult:string, EventType:string, EventSeverity:string)
+     ["push","Success","Install","Informational",
+      "create","Success","Create","Informational",
+      "delete","Success","Delete","Informational",
+      "repository","Success","Other","Informational",
+      "pull_request","Success","Other","Informational",
+      "membership","Success","Other","Informational",
+      "member","Success","Other","Informational",
+      "fork","Success","Create","Informational",
+      "installation","Success","Create","Informational",
+      "team","Success","Create","Informational",
+      "organization","Success","Other","Informational",
+      ];
+      githubscanaudit_CL
+     | where not(disabled)
+     | extend TimeGenerated = todatetime(tostring(split(TimeGenerated, '.', 0)[0]))
+      //  -- Pre filtering
+     | where 
+          (isnull(starttime) or TimeGenerated >= starttime) 
+          and (isnull(endtime) or TimeGenerated <= endtime)
+          and ((array_length(targetappname_has_any) == 0)) // targetappname not available on source
+          and ((array_length(srcipaddr_has_any_prefix) == 0))// ip address for source not available
+          and (array_length(srchostname_has_any) == 0) // SrcHostname not available in source
```

Comment:
incorrect filters used.

Reply:
updated

Reply:
you have missed configuring all filters, its mandatory to use all filters.